sympy/core/mul.py

```diff
@@ -1880,7 +1880,7 @@ def coeff_exp(term, x):
         try:
             for t in self.args:
                 coeff, exp = t.leadterm(x)
-                if not coeff.has(x):
+                if not coeff.has(x) and not coeff.has(sign) and not coeff.has(Abs):
```

Comment:
@jksuom is there a better way to write this? This condition is also the cause of #20697. If the condition is changed back to `if isinstance(coeff, Integer) or isinstance(coeff, Rational):` then both #9449 and #20697 are fixed, but 
```
def test_issue_13332():                                                         
    assert limit(sqrt(30)*5**(-5*x - 1)*(46656*x)**x*(5*x + 2)**(5*x + 5*S.Half) *
                (6*x + 2)**(-6*x - 5*S.Half), x, oo) == Rational(25, 36)
```
fails. I have been trying to look for a condition which works with all 3 of these but haven't been able to find one.

Reply:
Maybe something like `coeff.is_number`would work.